zapcore/sampler.go

```diff
@@ -38,6 +47,37 @@ type counter struct {
 
 type counters [_numLevels][_countersPerLevel]counter
 
+// Report defines a function that generates and writes sampling activity report
+// log entries.
+type Report func(core Core, ent Entry, n int64) error
+
+// SamplingReport is an example and the default implementation of Report type.
+// Refer to it when you want to customise reporting behaviour and message
+// composition.
+func SamplingReport(core Core, ent Entry, n int64) error {
+	if n <= 0 {
+		return nil
+	}
+	if !core.Enabled(ReportingLogLevel) {
+		return fmt.Errorf(
+			"reporting core must have at least %s logging level",
```

Comment:
Don't use `%s` and `x.String()` directly, just `fmt.Sprintf("%v", x)` (by any other alias, errorf in this case)